### ISSUE TITLE
Wrong average in metrics

### ISSUE THREAD

**Issue by uxen-ab** [CachetHQ/Cachet#2827]
Hello,

We begin with Cachet for our status page. We want to make a kind of "uptime", so every 5 minutes we send a request to the API with either 100 (%) if the website is up or 0 (%) if the website is down.
To do so I've created a metric named "uptime", and I choose "average".
I'm using the **2.3.9** version and it seems the average is wrong.

Example : I sent 3 times 100 because during 15 minutes our website is on. The average on the metric is 300 but it should be 100.
In the database there is only one _metric_point_ with value = 100 and counter = 3. It's nice, it allows to use less memory.

The weird thing is in the repositories (\CachetHQ\Cachet\Repositories\Metric\MySqlRepository for example), the SQL query is `AVG(value * counter)`. In our case, `value * counter` equals 300, but there is only one line in the table, si the average is 300. If I'm right, it should be something like `SUM(value * counter) / SUM(counter)`.

Am I right, or have I done something wrong?

**Comment by jbrooksuk** [comment]
Are you able to setup a 2.4 instance with duplicated data to test this on? Alternatively, would you be able to give me a DB dump of the `metrics` and `metric_points` tables.

**Comment by jbrooksuk** [comment]
Are you able to setup a 2.4 instance with duplicated data to test this on? Alternatively, would you be able to give me a DB dump of the `metrics` and `metric_points` tables.

**Comment by uxen-ab** [comment]
I've just setup an install. I'll join a zip containing two files. I've installed the 2.3.9 version, send 3 times "100" for a metric, and the average is 300% instead of 100%. Then I've upgraded to 2.4 (I'd also tried installing directly the 2.4, the result is the same) and the average is right. But I experience some other weird behaviours with the metrics. Should I create an other issue?
[cachet-metrics-bug.tar.gz](https://github.com/CachetHQ/Cachet/files/1526460/cachet-metrics-bug.tar.gz)
